examples/tests/invalid/phase_mismatch.test.w

```diff
@@ -0,0 +1,13 @@
+let my_func1 = (callback: inflight (num): num) => {  };
+my_func1((n: num): num => { return n * 2; });
+
+let my_func2 = (callback: (num): num) => {  };
+my_func2(inflight (n: num): num => { return n * 2; });
+
+interface IMyHandlerInterface1 {
+  inflight handle(n:  num): num;
+}
+
+let my_func3 = (callback: IMyHandlerInterface1) => {  };
+my_func3((n: num): num => { return n * 2; });
```

Comment:
What happens if the phases don't match and the argument types are incompatible? (can we add a test?)

Reply:
currently it would result in a type mismatch. phase mismatch is currently checked as "would the type match when we dont look at the phase"  instead of checking if the phases mismatch. Especially when we go the hint route we could simply check if the phase mismatch and add the hint, would make the check also easier.

Reply:
changed this to always show the hint when the expected types include no function with the given phase, thus also if the argument types mismatch. added a test for this.